src/xrpld/app/tx/detail/NFTokenModify.cpp

```diff
@@ -0,0 +1,90 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/tx/detail/NFTokenModify.h>
+#include <xrpld/app/tx/detail/NFTokenUtils.h>
+#include <xrpld/ledger/View.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/Rate.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+NFTokenModify::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureNonFungibleTokensV1_1) ||
+        !ctx.rules.enabled(featureDynamicNFT))
+        return temDISABLED;
+
+    if (NotTEC const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (auto owner = ctx.tx[~sfOwner]; owner == ctx.tx[sfAccount])
+        return temMALFORMED;
+
+    if (auto uri = ctx.tx[~sfURI])
+    {
+        if (uri->length() == 0 || uri->length() > maxTokenURILength)
+            return temMALFORMED;
+    }
+
+    return preflight2(ctx);
+}
+
+TER
+NFTokenModify::preclaim(PreclaimContext const& ctx)
+{
+    AccountID const account = ctx.tx[sfAccount];
+    AccountID const owner =
+        ctx.tx[ctx.tx.isFieldPresent(sfOwner) ? sfOwner : sfAccount];
+
+    if (!nft::findToken(ctx.view, owner, ctx.tx[sfNFTokenID]))
+        return tecNO_ENTRY;
+
+    // Check if the NFT is mutable
+    if (!(nft::getFlags(ctx.tx[sfNFTokenID]) & nft::flagMutable))
+        return tecNO_PERMISSION;
+
+    // Verify permissions for the issuer
+    if (AccountID const issuer = nft::getIssuer(ctx.tx[sfNFTokenID]);
+        issuer != account)
+    {
+        if (auto const sle = ctx.view.read(keylet::account(issuer)); !sle)
+            return tecNO_PERMISSION;  // Should not be possible
```

Comment:
Shouldn't this have a different error (like `tecINTERNAL` or `tefEXCEPTION`) if this shouldn't be possible?

Reply:
fixed